### ISSUE TITLE
Program killed when opening large TIFF image

### ISSUE THREAD

**Issue by jxu** [python-pillow/Pillow#3904]
### What did you do?

```python
from PIL import Image
Image.MAX_IMAGE_PIXELS = 1 << 36
im = Image.open("rawdata/images/111249.svs")
im = im.resize((im.width // 5, im.height // 5))
im.save("resize_images_test/111249.jpeg")
```

The TIFF file is very large (1 GB, 205176x85022)

The program is always killed, likely for memory usage.

### What are your OS, Python and Pillow versions?

* OS: Red Hat Enterprise Linux Server 7.6 (Maipo)
* Python: Python 3.7.1 (default, Dec 14 2018, 19:28:38)                                   
 [GCC 7.3.0] :: Anaconda, Inc. on linux
* Pillow: 6.0.0
* System: Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz, 64 GB RAM

**Comment by radarhere** [comment]
I suppose you don't want to link to the image because it is so large.

Could you let us know the value of `im.mode`?

Also, could you check at what point it dies?
- Does it die without the last line?
```python
from PIL import Image
Image.MAX_IMAGE_PIXELS = 1 << 36
im = Image.open("rawdata/images/111249.svs")
im = im.resize((im.width // 5, im.height // 5))
```
- Does it die loading, instead of resizing?
```python
from PIL import Image
Image.MAX_IMAGE_PIXELS = 1 << 36
im = Image.open("rawdata/images/111249.svs")
im.load()
```

**Comment by wiredfool** [comment]
Pretty sure that there are some limits to the number of pixels in an image related to some 32 bit values in the storage system, iirc somewhere in the range of 2gigapixels.

**Comment by cgohlke** [comment]
From the file name extension, this seems to be an [Aperio](https://openslide.org/formats/aperio/) pyramidal, tiled, JPEG or JPEG 2000 compressed whole-slide image. The largest pyramidal level would be ~49 GB uncompressed. That is if the decoding of the not quite TIFF compliant file succeeds. [Openslide-python](https://openslide.org/api/python/) is a specialized library for processing such files with minimum memory usage.

**Comment by jxu** [comment]
I've found success using vips to read and resize such images. Ideally Pillow can use virtual memory like ImageMagick or processing chunks of the image at a time like vips.
I'll try posting some example images soon.

**Comment by cgohlke** [comment]
I actually meant to recommend [pyvips](https://pypi.org/project/pyvips/) which is more up to date than  openslide-python and uses OpenSlide internally to read whole-slide images.

Maybe you could use a higher pyramidal level for resizing, e.g. the third level for reducing by 5. I had no issue resizing a 78000x30462 SVS image with Pillow 6 on Windows.

**Comment by wiredfool** [comment]
The new memory allocator may have improved some of the limits mentioned here: https://github.com/python-pillow/Pillow/blob/master/docs/reference/limits.rst

**Comment by jxu** [comment]
@radarhere I'm pretty sure it's just because Pillow is reading the whole file into memory. OOM happens when I try `im.load()` too. Same problem occurs on my Ubuntu 18.04 laptop. 

Here are some big test images made with VIPS (warning: images uncompressed are a few gigabytes). 

[big_images.zip](https://github.com/python-pillow/Pillow/files/3294577/big_images.zip)
![50000](https://user-images.githubusercontent.com/7989982/59569655-b767cb80-905a-11e9-9e5f-a41ca5e0a7d9.png)

**Comment by cgohlke** [comment]
I have no issue loading `100000.tiff` with Pillow 6 on Windows. The question is whether Pillow just quits/segfaults or raises a `MemoryError` in case not enough memory is available.

**Comment by jxu** [comment]
What are your computer specs?

**Comment by TheJKFever** [comment]
I was facing a similar issue, and since I was inside a dockerized container it was simple to increase the memory. I increased it from 2GB to 8GB and was then able to read the image fine. It just took over 2 minutes 😱 to do so...

Using `memory_profiler`, I was able to see exactly how much memory was being used just by opening the image. Here's the before and after:
`Memory usage (MB): [59.98046875, 4324.83984375]`
